software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessDriver.java

```diff
@@ -147,8 +147,18 @@ public void start() {
             }});
 
             DynamicTasks.queue("customize", new Runnable() { public void run() {
-                waitForConfigKey(BrooklynConfigKeys.CUSTOMIZE_LATCH);
-                customize();
+                DynamicTasks.queue("pre-customize-command", new Runnable() { public void run() {
+                    runPreCustomizeCommand();
```

Comment:
I'm not sure how I feel about only waiting for this customise latch in the main task. I think I would expect the latch to be obtained before `pre-customize-command` is run too.